src/main/java/net/dv8tion/jda/api/entities/RoleConnectionMetadata.java

```diff
@@ -0,0 +1,398 @@
+     *
+     * <p>This change will not take effect in Discord until you update the role connection metadata using {@link JDA#updateRoleConnectionMetadata(Collection)}.
+     *
+     * @param  map
+     *         The map from which to transfer the translated descriptions
+     *
+     * @throws IllegalArgumentException
+     *         <ul>
+     *             <li>If the map is null</li>
+     *             <li>If the map contains an {@link DiscordLocale#UNKNOWN} key</li>
+     *             <li>If the map contains a description which is empty or more than {@value MAX_DESCRIPTION_LENGTH} characters long</li>
+     *         </ul>
+     *
+     * @return This updated record instance
+     */
+    @Nonnull
+    public RoleConnectionMetadata setDescriptionLocalizations(@Nonnull Map<DiscordLocale, String> map)
+    {
+        this.descriptionLocalization.setTranslations(map);
+        return this;
+    }
+
+    @Override
+    public String toString()
+    {
+        return new EntityString(this)
+            .setType(type)
+            .addMetadata("key", key)
+            .addMetadata("name", name)
+            .toString();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (!(obj instanceof RoleConnectionMetadata))
+            return false;
+        if (this == obj)
+            return true;
+        RoleConnectionMetadata o = (RoleConnectionMetadata) obj;
+        return this.type == o.type
+            && this.key.equals(o.key)
+            && this.name.equals(o.name)
+            && this.description.equals(o.description);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(type, key, name, description);
+    }
+
+    @Nonnull
+    @Override
+    public DataObject toData()
+    {
+        return DataObject.empty()
+            .put("type", type.value)
+            .put("name", name)
+            .put("key", key)
+            .put("description", description)
+            .put("name_localizations", nameLocalization)
+            .put("description_localizations", descriptionLocalization);
+    }
+
+    /**
+     * Parses a {@link RoleConnectionMetadata} from a {@link DataObject}.
+     * <br>This is the reverse of {@link #toData()}.
+     *
+     * @param  data
+     *         The data object to parse values from#
+     *
+     * @throws IllegalArgumentException
+     *         If the provided data object is null
+     * @throws net.dv8tion.jda.api.exceptions.ParsingException
+     *         If the provided data does not have a valid int type value
+     *
+     * @return The parsed metadata instance
+     */
+    @Nonnull
+    public static RoleConnectionMetadata fromData(@Nonnull DataObject data)
+    {
+        Checks.notNull(data, "Data");
+        return new RoleConnectionMetadata(
+            MetadataType.fromValue(data.getInt("type")),
+            data.getString("name", null),
+            data.getString("key", null),
+            data.getString("description", null)
```

Comment:
Do we not get the localization maps back?

Reply:
This seems like a problem because of retrieve (GET) and update (PUT) because the localization maps will be lost.

Reply:
I might've forgotten adding it here